lib/bluetooth/ble_adv/ble_adv.c

```diff
@@ -91,49 +91,49 @@ static void on_disconnected(struct ble_adv *ble_adv, ble_evt_t const *ble_evt)
 
 static void on_terminated(struct ble_adv *ble_adv, ble_evt_t const *ble_evt)
 {
-	int err;
+	uint32_t nrf_err;
 	const uint8_t reason = ble_evt->evt.gap_evt.params.adv_set_terminated.reason;
 	struct ble_adv_evt adv_evt;
 
 	/* Start advertising in the next mode */
 	if (reason == BLE_GAP_EVT_ADV_SET_TERMINATED_REASON_TIMEOUT ||
 	    reason == BLE_GAP_EVT_ADV_SET_TERMINATED_REASON_LIMIT_REACHED) {
 		LOG_DBG("Advertising timeout");
-		err = ble_adv_start(ble_adv, adv_mode_next(ble_adv->mode_current));
-		if (err) {
-			adv_evt.error.reason = err;
+		nrf_err = ble_adv_start(ble_adv, adv_mode_next(ble_adv->mode_current));
+		if (nrf_err != NRF_SUCCESS) {
+			adv_evt.error.reason = nrf_err;
 			adv_evt.evt_type = BLE_ADV_EVT_ERROR;
 			ble_adv->evt_handler(ble_adv, &adv_evt);
 		}
 	}
 }
 
-static int flags_set(struct ble_adv *ble_adv, uint8_t flags)
+static uint32_t flags_set(struct ble_adv *ble_adv, uint8_t flags)
 {
-	int err;
+	uint32_t nrf_err;
 	uint8_t *parsed_flags;
 
 	parsed_flags = ble_adv_data_parse(ble_adv->adv_data.adv_data.p_data,
 					  ble_adv->adv_data.adv_data.len, BLE_GAP_AD_TYPE_FLAGS);
 
 	if (!parsed_flags) {
 		LOG_WRN("Unable to find flags in current advertising data");
-		return -EINVAL;
+		return NRF_ERROR_INVALID_PARAM;
 	}
 
 	*parsed_flags = flags;
 
-	err = sd_ble_gap_adv_set_configure(&ble_adv->adv_handle, &ble_adv->adv_data,
+	nrf_err = sd_ble_gap_adv_set_configure(&ble_adv->adv_handle, &ble_adv->adv_data,
 					   &ble_adv->adv_params);
```

Comment:
Check parameter alignment where `err` have been swapped with `nrf_err`.
```suggestion
	nrf_err = sd_ble_gap_adv_set_configure(&ble_adv->adv_handle, &ble_adv->adv_data,
					       &ble_adv->adv_params);
```